recipes/qt/6.x.x/conanfile.py

```diff
@@ -613,16 +613,16 @@ def generate(self):
         current_cpp_std = self.settings.get_safe("compiler.cppstd", default_cppstd(self))
         current_cpp_std = str(current_cpp_std).replace("gnu", "")
         cpp_std_map = {
-            11: "FEATURE_cxx11",
-            14: "FEATURE_cxx14",
-            17: "FEATURE_cxx17",
-            20: "FEATURE_cxx20"
+            "11": "FEATURE_cxx11",
+            "14": "FEATURE_cxx14",
+            "17": "FEATURE_cxx17",
+            "20": "FEATURE_cxx20"
         }
         if Version(self.version) >= "6.5.0":
-            cpp_std_map[23] = "FEATURE_cxx2b"
+            cpp_std_map["23"] = "FEATURE_cxx2b"
 
         for std, feature in cpp_std_map.items():
-            tc.variables[feature] = "ON" if int(current_cpp_std) >= std else "OFF"
```

Comment:
the comparison as it currently is, is correct, if I'm not mistaken

`current_cpp_std` is a string, and it is being cast to an `int`, and is being compared with an int (each of the `keys` in the `cpp_std_map` dictionary.

Testing this locally seems to behave as expected.